server/datastore/mysql/apple_mdm.go

```diff
@@ -4789,3 +4790,351 @@ WHERE
 		tok.ID)
 	return ctxerr.Wrap(ctx, err, "updating abm_token")
 }
+
+func (ds *Datastore) InsertVPPToken(ctx context.Context, tok *fleet.VPPTokenData, teamID *uint, nullTeam fleet.NullTeamType) (*fleet.VPPTokenDB, error) {
+	insertStmt := `
+	INSERT INTO
+		vpp_tokens (
+			organization_name,
+			location,
+			renew_at,
+			token,
+			team_id,
+			null_team_type
+		)
+	VALUES (?, ?, ?, ?, ?, ?)
+`
+
+	if teamID != nil && nullTeam != fleet.NullTeamNone {
+		return nil, ctxerr.Errorf(ctx, "nullTeam must be set to NullTeamNone if teamID is present")
+	}
+
+	if err := ds.checkVPPNullTeam(ctx, nil, nullTeam); err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "checking vpp token null team")
+	}
+
+	tokRawBytes, err := base64.StdEncoding.DecodeString(tok.Token)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "decoding raw vpp token data")
+	}
+
+	var tokRaw fleet.VPPTokenRaw
+	if err := json.Unmarshal(tokRawBytes, &tokRaw); err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "unmarshalling raw vpp token")
+	}
+
+	exp, err := time.Parse(fleet.VPPTimeFormat, tokRaw.ExpDate)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "parsing vpp token expiration date")
+	}
+	exp = exp.UTC()
+
+	tokEnc, err := encrypt([]byte(tok.Token), ds.serverPrivateKey)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "encrypt token with datastore.serverPrivateKey")
+	}
+
+	vppTokenDB := &fleet.VPPTokenDB{
+		OrgName:   tokRaw.OrgName,
+		Location:  tok.Location,
+		RenewDate: exp,
+		Token:     tok.Token,
+		TeamID:    teamID,
+		NullTeam:  nullTeam,
+	}
+
+	res, err := ds.writer(ctx).ExecContext(
+		ctx,
+		insertStmt,
+		vppTokenDB.OrgName,
+		vppTokenDB.Location,
+		exp,
+		tokEnc,
+		vppTokenDB.TeamID,
+		vppTokenDB.NullTeam,
+	)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "inserting vpp token")
+	}
+
+	id, err := res.LastInsertId()
```

Comment:
Nit: for our mysql driver, this can never fail so it can save a bit of boilerplate (`id, _ := res.LastInsertId()`).